core/src/main/java/org/mule/api/registry/LifecycleRegistry.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.registry;
+
+import org.mule.api.MuleException;
+
+/**
+ * A {@link Registry} which not only registers and fetches objects,
+ * but it's also capable of applying lifecycle and injects dependencies
+ *
+ * @since 3.7.0
+ */
+public interface LifecycleRegistry extends Registry
```

Comment:
haha... LifecycleRegistry is the name I gave it after @dfeist made a similar comment. Any advice on how to call it? I'm all open to suggestions